static/js/ui_init.js

```diff
@@ -231,6 +231,17 @@ $(function () {
         $(this).tooltip('hide');
     });
 
+    // Refocus the compose area after clicking links
+    $("body").on("focus", '*', function (e) {
+        if (e.target.id === 'new_message_content') {
+            $(".messagebox-content a").off("click.refocus").on("click.refocus", function () {
+                $("#new_message_content").focus();
+            });
+        } else if (e.target.tagName !== 'A') {
+            $(".messagebox-content a").off("click.refocus");
+        }
+    });
```

Comment:
The frequent live-editing of click handlers here feels scary to me (I have nightmares of hard-to-debug problems caused by this sort of thing).  I wonder if there's a way that we can prevent focus to going to the link in the first place when it's clicked?  I feel like that might be a more promising strategy.